### DISCUSSION TITLE
Configuring and keeping ranges for helm v3 on maintenance branch

### DISCUSSION THREAD

**Discussion by hpschry** [renovatebot/renovate#11487] [Request Help]
Hi, 
I have difficulties to configure ranges for maintenance branches when applying renovate on helm v3. The scenario is, that I would like to update to the latest version on the main branch, but on a maintenance branch I'd like to allow only updates within the range specified on the maintenance branch. 
Chart.yaml example section: 
```
dependencies:
  - name: frontend
    version: ~1.255.1
```
I'm using the following configuration: 
```javascript
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "commitMessagePrefix": "DEVOPS-681 fix:",
  "baseBranches": [
    "master",
    "release/ft10/1.9.x"
  ],
  "rangeStrategy": "bump", 
  "packageRules": [
    {
      "matchManagers": [ "helmv3" ], 
      "groupName": "ICE helm charts",
      "automerge": true
    },
    {
      "matchManagers": [ "helmv3" ],
      "matchBaseBranches": [ "release/ft10/1.9.x" ],
      "groupName": "ICE FT10 helm charts",
      "automerge": true,
      "versioning": "npm",
      "rangeStrategy": "bump",
      "schedule": "at any time"
    }
  ]
}
```
`release/ft10/1.9.x` is our maintenance branch. I configured npm for the versioning scheme, because it seems to me that this is similar to what helm accepts.
I would have expected, that the bump strategy bumps to the highest available version within the 1.255.x range, but it always inserts the highest available version, e.g. 1.256.8, which is outside of the maintenance range. 
The optimal configuration would be, that the range is kept as is, and only the lockfile would get updated, however I also failed to configure this other scenario either.
I'm wondering what i may have missed in the configuration. Thanks for any pointers.

**Comment by viceice** [comment]
Then you should disable major, minor or patch updates on your maintenance branch and enable lockfileMaintenance

https://docs.renovatebot.com/configuration-options/#lockfilemaintenance

**Reply by viceice** [reply]
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "commitMessagePrefix": "DEVOPS-681 fix:",
  "baseBranches": [
    "master",
    "release/ft10/1.9.x"
  ],
  "rangeStrategy": "bump", 
  "packageRules": [
    {
      "matchManagers": [ "helmv3" ], 
      "groupName": "ICE helm charts",
      "automerge": true
    },
    {
      "matchManagers": [ "helmv3" ],
      "matchBaseBranches": [ "release/ft10/1.9.x" ],
      "matchUpdateTypes": ["major", "minor", "patch" ],
      "enabeld": false
    },
    {
      "matchManagers": [ "helmv3" ],
      "matchBaseBranches": [ "release/ft10/1.9.x" ],
      "lockFileMaintenance": { "enabled": true }
    }
  ]
}
```

**Comment by hpschry** [comment]
Thanks for the quick answer!
I tried to apply your proposed configuration, but get now: "message": "packageRules[1]: Each packageRule must contain at least one non-match* or non-exclude* field. Rule: {\"matchManagers\":[\"helmv3\"],\"matchBaseBranches\":[\"release/ft10/1.9.x\"]}"
For debugging purposes I stripped the config down to: 
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "commitMessagePrefix": "DEVOPS-681 fix:",
  "baseBranches": [
    "master",
    "release/ft10/1.9.x"
  ],
  "rangeStrategy": "bump", 
  "packageRules": [
    {
      "matchManagers": [ "helmv3" ], 
      "groupName": "ICE helm charts",
      "automerge": true
    },
    {
      "matchManagers": [ "helmv3" ],
      "matchBaseBranches": [ "release/ft10/1.9.x" ],
      "lockFileMaintenance": { "enabled": true },
      "schedule": "at any time"
    }
  ]
}
```
Before I tried with the complete config.

**Reply by rarkins** [reply]
@Viceice it might be config massaging gone wrong. We generally don't want nested objects within packageRules

**Reply by viceice** [reply]
@rarkins Whats the solution? 🤔 not sure how to resolve the lockfile maintenance config.

**Reply by rarkins** [reply]
Here's how it massages today:

```
Object {
  "packageRules": Array [
    Object {
      "matchBaseBranches": Array [
        "release/ft10/1.9.x",
      ],
      "matchManagers": Array [
        "helmv3",
      ],
      "schedule": Array [
        "at any time",
      ],
    },
    Object {
      "enabled": true,
      "matchBaseBranches": Array [
        "release/ft10/1.9.x",
      ],
      "matchManagers": Array [
        "helmv3",
      ],
      "matchUpdateTypes": Array [
        "lockFileMaintenance",
      ],
    },
  ],
}
```

i.e. it's treating `lockFileMaintenance` as a matching field and not as a result.

**Reply by rarkins** [reply]
#11522
